app/tournament/tournament.js

```diff
@@ -0,0 +1,101 @@
+import * as player from "./player";
+import * as match from "./match";
+import * as utils from "../common/utils";
+
+export class Tournament {
+    constructor() {
+        this.players = [];
+        this.matches = [];
+    }
+
+    addPlayer(player) {
+        if (this.existsAlready(player)) {
+            throw new TournamentError(`Duplicate player: "${player.name}".`);
```

Comment:
So after adding some code validation outside, should I clear my exceptions off my modules or just the try/catch blocks?